rawfile/filesystem/btrfs.py

```diff
@@ -0,0 +1,180 @@
+import subprocess
+from .utils import get_device_for_mountpoint
+
+
+class BTRFS(FileSystemBase):
+    @property
+    def __filesystem__(self) -> str:
+        return "btrfs"
+
+    def format_fs(self, options: list[str] = []):
+        try:
+            with TemporaryDirectory(
+                prefix=f"rawfile.{self.__filesystem__}.fs-bootstrap."
+            ) as tmp_dir:
+                try:
+                    default_subvol = f"{tmp_dir}/default"
+                    output = subprocess.run(
+                        f"""
+                        set -ex
+                        mkfs.{self.__filesystem__} {" ".join(options)} {self.device}
+                        mount -t btrfs {self.device} {tmp_dir}
+                        btrfs subvolume create {default_subvol}
+                        btrfs subvolume set-default {default_subvol}
+                        """,
+                        shell=True,
+                        check=True,
+                        capture_output=True,
+                    )
+                finally:
+                    subprocess.run(
+                        f"umount {tmp_dir}",
+                        shell=True,
+                        check=True,
+                        capture_output=True,
+                    )
+
+            return output.stdout.decode()
+        except subprocess.CalledProcessError as e:
+            raise FileSystemFormatError(self.__filesystem__, e.stderr.decode())
+        except Exception as e:
+            raise FileSystemFormatError(self.__filesystem__, str(e))
+
+    def mount(
+        self,
+        mountpoint: str | None = None,
+        options: list[str] = [],
+    ) -> str:
+        if mountpoint is None:
+            if self.mountpoint is None:
+                raise InvalidMountpointError()
+            mountpoint = self.mountpoint
+        options.extend(["-o", "flushoncommit"])
+        self.mountpoint = mountpoint
+        try:
+            current_dev = get_device_for_mountpoint(mountpoint)
+            if current_dev == self.device:
+                return f"Device {self.device} already mounted at {mountpoint}"
+            elif current_dev:
+                self.unmount()
+            Path(mountpoint).mkdir(exist_ok=True)
+            output = subprocess.run(
+                f"mount -t {self.__filesystem__} {' '.join(options)} {self.device} {mountpoint}",
+                shell=True,
+                check=True,
+                capture_output=True,
+            )
+            return output.stdout.decode()
+        except subprocess.CalledProcessError as e:
+            raise FileSystemMountError(self.__filesystem__, e.stderr.decode())
+        except Exception as e:
+            raise FileSystemMountError(self.__filesystem__, str(e))
+
+    def unmount(self, clear_mountpoint=False) -> str:
+        try:
+            if get_device_for_mountpoint(self.mountpoint) != self.device:
+                return f"{self.device} is not mounted at {self.mountpoint}"
```

Comment:
if `clear_mountpoint` then we should also rmdir the mountpoint here for idempotency

Reply:
Already done at rawfile/filesystem/btrfs.py:96

Reply:
I mean if `get_device_for_mountpoint` returns `None`
This can happen in umount call gets stuck for a long time and csi-node pod is deleted

Reply:
Oh, yeah it would not remove that when we get into this situation, will move that to finally so it will run anytime